test/sendSpec.js

```diff
@@ -39,44 +43,60 @@ describe('Send', () => {
     options._store = new ObjStore()
     this.plugin = new PluginPaymentChannel(options)
 
+    this.mockSocket = new MockSocket()
+    this.plugin.addSocket(this.mockSocket)
     yield this.plugin.connect()
+
+    this.ilpError = ilpPacket.serializeIlpError({
+      code: 'F00',
+      name: 'Bad Request',
+      triggeredBy: '',
+      forwardedBy: [],
+      triggeredAt: new Date(),
+      data: JSON.stringify({ message: 'Peer isn\'t feeling like it.' })
+    })
   })
 
   afterEach(function * () {
-    assert(nock.isDone(), 'nocks should all have been called')
+    assert(yield this.mockSocket.isDone(), 'request handlers must have been called')
   })
 
   describe('RPC', () => {
     it('should throw an error on an error code', function () {
-      nock('https://example.com')
-        .post('/rpc?method=method&prefix=example.red.', [])
-        .reply(500)
+      const expectedRequestId = 1234
+      this.mockSocket.reply(clpPacket.TYPE_MESSAGE, ({requestId, data}) => {
+        assert.equal(requestId, expectedRequestId)
+        return clpPacket.serializeError({rejectionReason: this.ilpError}, requestId, [])
+      })
 
-      return expect(this.plugin._rpc.call('method', 'example.red.', []))
+      const cllpMessage = clpPacket.serializeMessage(expectedRequestId, [])
+      return expect(this.plugin._rpc._call(expectedRequestId, cllpMessage))
         .to.eventually.be.rejected
     })
 
-    it('should send authorization bearer token', function () {
-      nock('https://example.com', {
-        reqheaders: {
-          'Authorization': 'Bearer ' + this.plugin._getAuthToken()
-        }
-      })
-        .post('/rpc?method=method&prefix=example.red.', [])
-        .reply(200, { a: 'b' })
+    // TODO: reassess whether this test case is still necessary (cc: sharafian)
```

Comment:
we can re-write this test once we've settled on how authentication works for this plugin